app/helpers/needs_helper.rb

```diff
@@ -1,14 +1,22 @@
 module NeedsHelper
   def populate_needs
+    rejected = []
+    approved = []
     session[:cart].each do |id, qty|
-      need = NeedType.find(id)
-      current_user.needs.create(name: pluralize(qty, need.name),
-                                description: need.description,
-                                cost: need.cost * qty,
-                                image_url: need.image_url,
-                                slug: need.slug,
-                                category: need.category
-                                )
+      need_type = NeedType.find(id)
+      if current_user.needs.exists?(slug: need_type.slug)
+        rejected << need_type.name
+      else
+        current_user.needs.create(name: pluralize(qty, need_type.name),
+                                  description: need_type.description,
+                                  cost: need_type.cost * qty,
+                                  image_url: need_type.image_url,
+                                  slug: need_type.slug,
+                                  category: need_type.category
+                                  )
+        approved << need_type.name
```

Comment:
ok. i like the use of the helper module to abstract out and remove some of the work out of the controller.  i think that we were worried about using .create method yesterday, in case, something like duplicate amount or more than max amount was placed into their cart, but i think that having a check like this works, so that if it passes the check, then the method can go ahead and create the needs.  ... since we the priority is about getting this to be functional at this point in time, i'll go ahead and merge it to master.    later when we have our apps working, i imagine we want to create a method so that all need type attributes we are passing into the controller can be moved out to a helper method.  btw, now that the category is its own table, and an association, and not an attribute of needs, something will break.  but it should be an easy fix.  
thanks for your hard work jared. there was a lot of logic you had to get through to get this to work, i'm sure. 👍